osf/features.py

```diff
@@ -1,38 +1,6 @@
-DISABLE_ENGAGEMENT_EMAILS = 'disable_engagement_emails'
-ELASTICSEARCH_METRICS = 'elasticsearch_metrics'
-STORAGE_USAGE = 'storage_usage'
-ENFORCE_CSRF = 'enforce_csrf'
-INSTITUTIONAL_LANDING_FLAG = 'institutions_nav_bar'
-STORAGE_I18N = 'storage_i18n'
-OSF_PREREGISTRATION = 'osf_preregistration'
-OSF_GROUPS = 'osf_groups'
-ENABLE_CHRONOS = 'enable_chronos'
-EGAP_ADMINS = 'egap_admins'
+import yaml
 
-EMBER_AB_TESTING_HOME_PAGE_VERSION_B = 'ab_testing_home_page_version_b'
-EMBER_AB_TESTING_HOME_PAGE_HERO_TEXT_VERSION_B = 'ab_testing_home_page_hero_text_version_b'
-EMBER_AUTH_REGISTER = 'ember_auth_register'
-EMBER_CREATE_DRAFT_REGISTRATION = 'ember_create_draft_registration_page'
-EMBER_EDIT_DRAFT_REGISTRATION = 'ember_edit_draft_registration_page'
-EMBER_FILE_DETAIL = 'ember_file_detail_page'
-EMBER_MEETINGS = 'ember_meetings_page'
-EMBER_MEETING_DETAIL = 'ember_meeting_detail_page'
-EMBER_MY_PROJECTS = 'ember_my_projects_page'
-EMBER_PROJECT_ANALYTICS = 'ember_project_analytics_page'
-EMBER_PROJECT_CONTRIBUTORS = 'ember_project_contributors_page'
-EMBER_PROJECT_DETAIL = 'ember_project_detail_page'
-EMBER_PROJECT_FILES = 'ember_project_files_page'
-EMBER_PROJECT_FORKS = 'ember_project_forks_page'
-EMBER_PROJECT_REGISTRATIONS = 'ember_project_registrations_page'
-EMBER_PROJECT_SETTINGS = 'ember_project_settings_page'
-EMBER_PROJECT_WIKI = 'ember_project_wiki_page'
-EMBER_REGISTRATION_FORM_DETAIL = 'ember_registration_form_detail_page'
-EMBER_REGISTRIES_DETAIL_PAGE = 'ember_registries_detail_page'
-EMBER_SEARCH_PAGE = 'ember_search_page'
-EMBER_USER_PROFILE = 'ember_user_profile_page'
-EMBER_USER_SETTINGS = 'ember_user_settings_page'
-EMBER_USER_SETTINGS_ACCOUNTS = 'ember_user_settings_account_page'
-EMBER_USER_SETTINGS_ADDONS = 'ember_user_settings_addons_page'
-EMBER_USER_SETTINGS_APPS = 'ember_user_settings_apps_page'
-EMBER_USER_SETTINGS_NOTIFICATIONS = 'ember_user_settings_notifications_page'
-EMBER_USER_SETTINGS_TOKENS = 'ember_user_settings_tokens_page'
+with open('osf/features.yaml') as yaml_file:
+    yaml_data = yaml.load(yaml_file, Loader=yaml.FullLoader)
+    switches = yaml_data['switches']
```

Comment:
I hate to disagree with first round of code review, but this change has upped the risk of the ticket significantly since you've touched so much more of the codebase, and I'm not sure if it's gotten us much in return. I'm not opposed to the yaml inherently (though how is the yaml better than just storing a dictionary or two inside of this file?), but if we leave the imports from the rest of the code the way they were, then the change is much smaller.

I can be talked out of changing this again, but I think we'd have to increase QA's work somewhat if we want to keep it like this.

Reply:
I tend to agree that I don't see the equivalent benefit to the amount of risk incurred. @Johnetordoff what is your stance?

Reply:
Okay this is actually pretty weird but I really don't want to have to check for dunders, or change many files so here's my suggested middle ground:
```
flags = {
    'DISABLE_ENGAGEMENT_EMAILS': 'disable_engagement_emails',
    'ELASTICSEARCH_METRICS': 'elasticsearch_metrics',
    'STORAGE_USAGE': 'storage_usage',
    'ENFORCE_CSRF': 'enforce_csrf',
    'INSTITUTIONAL_LANDING_FLAG': 'institutions_nav_bar',
    'STORAGE_I18N': 'storage_i18n',
    'OSF_PREREGISTRATION': 'osf_preregistration',
    'OSF_GROUPS': 'osf_groups',
    'ENABLE_CHRONOS': 'enable_chronos',
    'EGAP_ADMINS': 'egap_admins'
}
locals().update(flags)
```
Then just import flags or possibly combine that with the YAML?